shared/desktop/app/ctl.desktop.js

```diff
@@ -12,6 +12,10 @@ export function ctlStop(callback: any) {
     args = ['ctl', 'stop']
     plat = 'win32'
   }
+  if (isLinux) {
```

Comment:
why isn't there an RPC to do this? Seems like calling out to the CLI is pretty gross. Just wondering about why this happened.

Reply:
No idea, just followed pattern. @zanderz do you know?

Reply:
Feel really brittle to shell out here

Reply:
Can you move the "Stopall" code into the service?  And the last thing it does is harakiri?

Reply:
what do you mean by move into the service? the `-all` flag? the last thing the js does is stop the process itself, but ctl stop should've killed it before - I think this is only for the case where ctl stop did not work, it still closes the gui anyway. 

Also, I wonder if the intent was so that it still works if somehow the rpc connection between the service and the gui were disconnected but the service still listens to cmdline rpcs. Don't know if that's likely.

Reply:
I don't think the RPC existed yet when this was put in, I'll take a look and refresh my memory. Note the updater does a blunt kill as well rather than an RPC I believe.

Reply:
wait, sorry, that doesn't make sense - of course `ctl.avdl` and so forth were there, maybe it should call that directly, like you say.

Reply:
maybe trying to call the RPC was just too scary? I just made a PR to do it this way but I don't think it quite works. Still don't have the hang of it after 2 years.  https://github.com/keybase/client/pull/16835